rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+
+
+#### User Actions:
+
+**Step 1**: **Smart Contract Selection and Input Creation**
+- **Smart Contract Selection**: The user selects a smart contract they wish to invoke.
+- **Input Creation**: The user creates a set of inputs required for the invocation by reading the necessary data from both the public and private states. This includes:
+  - Public data such as current account balances, public keys, and smart contract states.
+  - Private data such as private account balances and UTXOs.
+
+**Step 2**: **Choosing Execution Type**
+- **Execution Type Selection**: The user selects the type of execution based on their privacy needs. The options include:
+  - **Public Execution**: Suitable for transactions where transparency is desired.
+  - **Private Execution**: Used when transaction details need to be confidential.
+  - **Shielded Execution**: Hides the receiver's identity.
+  - **Deshielded Execution**: Hides the sender's identity.
+- **zkVM Requirement**: If the execution involves private, shielded, or deshielded types, the user must call the zkVM to handle these confidential transactions. 
+For purely public executions, the zkVM is not needed, and the user can directly transmit the transaction code to the sequencer.
+
+**Step 3**: **Calling zkVM for Proof Generation**
+- **zkVM Compilation**: The user calls the zkVM to compile the smart contract with both public and private inputs.
+  - **Kernel Circuit Proofs**: The zkVM generates individual proofs for each execution type through kernel circuits.
+  - **Proof Aggregation**: The zkVM aggregates these individual proofs into a single comprehensive proof, combining both private and public inputs.
+
+**Step 4**: **Transmitting Public Inputs and Retaining Private Inputs**
+- **Retaining Private Inputs**: The user keeps the private inputs secure and does not transmit them.
+- **Revealing Public Inputs**: The user transmits the following public inputs to the sequencer:
+  - Public inputs of the recursive proof
+  - Hashes of UTXOs
+  - Updates to the public state
+  - Transaction signature
+  - Nullifiers (to prevent double spending)
+
+After completing these steps, the user's part of the execution is done, and the sequencer takes over the process.
+
+#### Sequencer Actions:
+
+**Step 5**: **Proof Verification**
+- **Proof and Data Reception**: The sequencer receives the proof and public inputs from the user.
+- **Verification Process**:
+  - For private, shielded, and deshielded executions, the sequencer verifies the proof using the provided public data.
+  - For public executions, the sequencer reruns the smart contract code with the provided inputs to check the results.
+- **Validation**: If both the zkVM proofs and public execution results are verified successfully, the sequencer collects the proof and public data to proceed. 
+If verification fails, the process is aborted, and the transaction is rejected.
+
+**Step 6**: **Aggregating Proofs and Finalizing the Block**
```

Comment:
Replace:
"Aggregating Proofs and Finalizing the Block"

with:
"Aggregating proofs and finalizing the block"